pydantic/_hypothesis_plugin.py

```diff
@@ -223,7 +232,7 @@ def resolve_json(cls):  # type: ignore[no-untyped-def]
             extend=lambda x: st.lists(x) | st.dictionaries(st.text(), x),  # type: ignore
         )
     return st.builds(
-        json.dumps,
+        cls.inner_type.json if is_base_model(getattr(cls, 'inner_type', None)) else json.dumps,
```

Comment:
I think you can use [lenient_issubclass function from piydantic utils](https://github.com/samuelcolvin/pydantic/blob/50bc758495f5e22f3ad86d03bcf6c645a7d029cf/pydantic/utils.py#L176) here and we don't need the `is_base_model` function:

```suggestion
        cls.inner_type.json if lenient_issubclass(getattr(cls, 'inner_type', None), pydantic.BaseModel) else json.dumps,
```